File: test-suite/src/test/java/io/micronaut/guice/doc/examples/bindings/annotations/BindingAnnotationTest.java
package io.micronaut.guice.doc.examples.bindings.annotations;

import static org.junit.jupiter.api.Assertions.assertInstanceOf;

import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

@MicronautTest(startApplication = false)
class BindingAnnotationTest {
    @Inject @PayPal CreditCardProcessor paypalProcessor;
    @Inject @GoogleCheckout CreditCardProcessor checkoutProcessor;

    @Test
    void testInjectWithQualifiers() {
        assertInstanceOf(CheckoutCreditCardProcessor.class, checkoutProcessor);
        assertInstanceOf(PayPalCreditCardProcessor.class, paypalProcessor);
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/guice/doc/examples/bindings/annotations/BindingAnnotationTest.kt
package io.micronaut.guice.doc.examples.bindings.annotations

import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertInstanceOf
import org.junit.jupiter.api.Test

@MicronautTest(startApplication = false)
class BindingAnnotationTest {
    @Inject @PayPal lateinit var paypalProcessor: CreditCardProcessor
    @Inject @GoogleCheckout lateinit var checkoutProcessor: CreditCardProcessor

    @Test
    fun testInjectWithQualifiers() {
        assertInstanceOf(CheckoutCreditCardProcessor::class.java, checkoutProcessor)
        assertInstanceOf(PayPalCreditCardProcessor::class.java, paypalProcessor)
    }
}
